include/uxr/agent/middleware/ced/CedMiddleware.hpp

```diff
@@ -28,123 +28,339 @@ class CedMiddleware : public Middleware
+     */
     bool delete_subscriber(uint16_t subscriber_id) override;
 
+    /**
+     * @brief Removes a DataWriter from the datawriters register.
+     * @param datawriter_id The CedDataWriter identifier.
+     * @return  true in case of the CedDataWriter was found and removed, false in other case.
+     */
     bool delete_datawriter(uint16_t datawriter_id) override;
 
+    /**
+     * @brief Removes a DataReader from the datawriters register.
+     * @param datareader_id The CedDataReader identifier.
+     * @return  true in case of the CedDataReader was found and removed, false in other case.
+     */
     bool delete_datareader(uint16_t datareader_id) override;
 
-/**********************************************************************************************************************
- * Write/Read functions.
- **********************************************************************************************************************/
+    /**
+     * @brief Writes data using the CedDataWriter identified by the datawriter_id parameter.
+     * @param datawriter_id The CedDataWriter identifier.
+     * @param data          The data to be written.
+     * @return  true in case of successful writing and false in other case.
+     */
     bool write_data(
             uint16_t datawriter_id,
             std::vector<uint8_t>& data) override;
 
+    /**
+     * @brief Read data using the CedDataReader identified by the datareader_id paramenter.
+     *        This is a blocking function that will block at most "timeout" milleseconds.
+     * @param datareader_id The CedDataReader's identifier.
+     * @param data          The data read.
+     * @param timeout       The timeout (milliseconds) of the reading.
+     * @return  true in case of successful reading and false in other case.
+     */
     bool read_data(
             uint16_t datareader_id,
             std::vector<uint8_t>& data,
             std::chrono::milliseconds timeout) override;
 
-/**********************************************************************************************************************
- * Matched functions.
- **********************************************************************************************************************/
+    /**
+     * @brief Checks whether an existing CedParticipant, identified by the participant_id, matches with a new
+     *        CedParticipant that would result from the creation of a new one using the domain_id and the reference
+     *        representation.
+     *        It is considered that the CedParticipants match if both have the same domain_id.
+     * @param participant_id    The existing CedParticipant identifier.
+     * @param domain_id         The Domain identifier of the new CedParticipant.
+     * @param ref               The reference that represents the new CedParticipant (unused).
+     * @return  true if both CedParticipant have the same Domain identifier, and false in other case.
```

Comment:
This is true?
Do two participants match just matching the Domain?.

Reply:
Yes, in CedMiddleware it is true.